### PR TITLE
added gemini exp 1121 and 1206 to api.ts

### PR DESCRIPTION
**Pull request by rurounigit**
I added gemini exp 1121 to the api.ts 
according to/based on the original data provided by the google api:

{
      "name": "models/gemini-exp-1121",
      "version": "exp-1121",
      "displayName": "Gemini Experimental 1121",
      "description": "Experimental release (November 21st, 2024) of Gemini.",
      "inputTokenLimit": 32768,
      "outputTokenLimit": 8192,
      "supportedGenerationMethods": [
        "generateContent",
        "countTokens"
      ],
      "temperature": 1,
      "topP": 0.95,
      "topK": 64,
      "maxTemperature": 2
    },

### CODE REVIEW

File: src/shared/api.ts [hunk 1]
```diff
@@ -277,6 +277,16 @@ export const geminiModels = {
 		inputPrice: 0,
 		outputPrice: 0,
 	},
+
```
Comment (lloydchang) [inline src/shared/api.ts:280]:
```suggestion
```

File: src/shared/api.ts [hunk 2]
```diff
@@ -277,6 +277,16 @@ export const geminiModels = {
...
+        "gemini-exp-1121": {
+		maxTokens: 8192,
+                contextWindow: 32_768,
+                supportsImages: true,
+		supportsPromptCache: false,
+		inputPrice: 0,
+		outputPrice: 0,
+        },
+
```
Comment (lloydchang) [inline src/shared/api.ts]:
```suggestion
```

**Comment by lloydchang** [comment]
In my https://github.com/lloydchang/cline-gemini-exp-1121-space-invaders

---

I tried this pull request:

> [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-pro-exp-0827:streamGenerateContent?alt=sse: [429 Too Many Requests] Resource has been exhausted (e.g. check quota).

can happen periodically.

---

I believe there's a quota of 2 requests per minute.

---

According to https://discuss.ai.google.dev/t/whats-the-rate-limit-for-the-experimental-models/38226

> In your case (gemini-1.5-pro-exp-0827) Rate limits are 2 RPM, 32,000 TPM, 50 RPD.

---

**Comment by lloydchang** [comment]
To test this pull request, you can run:
```zsh
gh repo clone cline/cline
cd cline
gh pr checkout 851 --repo cline/cline
npm run install:all
npm install -g @vscode/vsce
vsce package
code --install-extension claude-dev-2.1.6.vsix
code
```

File: src/shared/api.ts (hunk 1 above)
Comment (lloydchang) [reply src/shared/api.ts:280]: suggestion: commit above suggested change to remove extraneous line — to match formatting of other entries in the same file, `api.ts`

File: src/shared/api.ts (hunk 2 above)
Comment (lloydchang) [reply src/shared/api.ts]: suggestion: commit above suggested change to remove extraneous line — to match formatting of other entries in the same file, `api.ts`

**Comment by lloydchang** [comment]
Subsequently, I'm amending https://github.com/lloydchang/aider-cline-gemini-exp-1121-space-invaders with Aider

**Comment by lloydchang** [comment]
> Subsequently, I'm amending https://github.com/lloydchang/aider-cline-gemini-exp-1121-space-invaders with Aider

---

Related information from Cline Discord's `#prompts` channel at https://discord.com/channels/1275535550845292637/1275555786621325382/1313115383070330921

---

Aider's `--watch-files` feature allows me to experiment with having Cline and Aider pair program with each other.

For example, I tasked Cline to pair program with Aider:

> To facilitate pair programming between Cline and Aider, please review the currently open file and add comments that begin with "// AI " followed by any programming instructions you would like Aider, an AI pair programming tool, to execute. Conclude with a final line "// AI!" Afterward, when Aider edits the file, do not allow Cline to make any changes.

1. Cline edits the file and asks Aider to pair program

2. Aider edits the file in response to Cline

The nice thing about Aider is that it is a command line tool that now runs in the background, so I can try experiments like the above pair programming exercise between Cline and Aider.

**Comment by lloydchang** [comment]
**Note to my future self:**

---

If / when Saoud and Cline team review [my pull requests for `cline/cline`](https://github.com/cline/cline/pulls?q=by%3A+lloydchang) and if revisions are needed, I'll try the following tactic:

---

**Ask Cline to:**

> write a test for the changes in `[sha]`

and

> write a test for feedback from Saoud and Cline team: `[feedback here]`

---

… to save me the effort from trying to re-remember what I did in pull requests from a long time ago.

**Comment by SaidTorres3** [comment]
where is gemini-2.0-flash-exp?!

**Comment by SaidTorres3** [comment]
Also, add gemini-2.0-flash-exp

**Comment by lloydchang** [comment]
@SaidTorres3 wrote:
> Also, add gemini-2.0-flash-exp

Alternatively, try [Roo Cline](https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline)

It provides `gemini-2.0-flash-exp  via @mrubens
* https://github.com/RooVetGit/Roo-Cline/pull/74
* https://marketplace.visualstudio.com/items/RooVeterinaryInc.roo-cline/changelog
> [2.1.20]
> Add Gemini 2.0
* https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline

**Comment by SaidTorres3** [comment]
yeah, I already installed it.

**Review by ivangermanov — APPROVED** [review]

**Comment by saoudrizwan** [comment]
Sorry for not adding this sooner–fixed in v3. Thank you for the PR

File: src/shared/api.ts (hunk 1 above)
Comment (saoudrizwan) [reply src/shared/api.ts:280]: Thanks for the input @lloydchang 🙏

**Pull request closed without merging**